examples/linear_model/plot_bayesianridge_curvefit.py

```diff
@@ -0,0 +1,82 @@
+"""
+============================================
+Curve Fitting with Bayesian Ridge Regression
+============================================
+
+Computes a Bayesian Ridge Regression of Sinusoids.
+
+See :ref:`bayesian_ridge_regression` for more information on the regressor.
+
+In general, when fitting a curve with a polynomial by Bayesian ridge
+regression (evidence approxymation), the selection of initial values of
+hyperparameters (alpha, lambda) may be important.
+
+In this example, the sinusoid is approximated by a polynomial using different
+pairs of initial values.
+
+When starting from the default values (alpha_init =1.90, lambda_init = 1.),
+the bias of the resulting curve is large, and the variance is small.
+So, lambda_init should be relatively small (1.e-3) so as to reduce the bias.
+
+Also, by evaluating the evidences (marginal log-likelihoods, logL) of
+these models, we can determine which one is better.
+It can be concluded that the model with larger evidence are more likely.
+"""
+print(__doc__)
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.linear_model import BayesianRidge
+
+
+def func(x): return np.sin(2*np.pi*x)
+
+
+# #############################################################################
+# Generate sinusoidal data with noise
+size = 25
+np.random.seed(1234)
+xtrain = np.random.uniform(0., 1., size)
+ytrain = func(xtrain)+np.random.normal(scale=0.1, size=size)
+xtest = np.linspace(0., 1., 100)
+
+
+# #############################################################################
+# Fit by cubic polynomial
+nOrder = 3
+Xtrain = np.vander(xtrain, nOrder+1, increasing=True)
+Xtest = np.vander(xtest, nOrder+1, increasing=True)
```

Comment:
please stick to the sklearn standard naming conventions:

n_order
X_train
X_test
x_train
x_test